AsyncDisplayKit/ASVideoNode.h

```diff
@@ -0,0 +1,34 @@
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+typedef NS_ENUM(NSUInteger, ASVideoGravity) {
+  ASVideoGravityResizeAspect,
+  ASVideoGravityResizeAspectFill,
+  ASVideoGravityResize
+};
+
+@protocol ASVideoNodeDelegate;
+
+@interface ASVideoNode : ASControlNode<_ASDisplayLayerDelegate>
+@property (atomic, strong, readwrite) AVAsset *asset;
+@property (atomic, strong, readonly) AVPlayer *player;
+@property (atomic, strong, readonly) AVPlayerItem *currentItem;
+@property (nonatomic, assign, readwrite) BOOL shouldAutoplay;
+@property (atomic) ASVideoGravity gravity;
+@property (atomic) BOOL autorepeat;
+@property (atomic) ASButtonNode *playButton;
+
+@property (atomic, weak, readwrite) id<ASVideoNodeDelegate> delegate;
+
+- (void)play;
+- (void)pause;
+
+- (BOOL)isPlaying;
```

Comment:
Maybe we should have something like isPlayable or isReadyToPlay.  If we added that - I guess it could be later, as autoplay is the most key feature here - we'd need a delegate method too that notifies when it becomes ready to play.  There should also be some option to control whether the play button is visible before it's ready to play (it might pulsate the alpha with an autoreversing animation if the video is still loading), or if the button should only become visible once it's ready to play.

I'm suggesting many of these things because I think you probably already have a running list of such feature ideas from your own ideas, not because all are required right now - indeed wrapping this up with stable behavior and a minimized API that we are fairly confident won't change (only grow) is the best goal, because of the huge value of the programming guide.  It would be awesome if you could write a GitHub Issue with a loosely organized list of these kind of feature requests combined with the ones you've thought of, so that community members can join the discussion there and decide on what the "2.0" of this component looks like - and find who will build it!